tchannel/tchannel.py

```diff
@@ -240,6 +243,12 @@ def advertise(self, routers, name=None, timeout=None):
             When unable to make our first advertise request to Hyperbahn.
             Subsequent requests may fail but will be ignored.
         """
+        if routers is None and router_file is not None:
+            # should just let the exceptions fly
+            with open(router_file, 'r') as json_data:
+                routers = json.load(json_data)
+                json_data.close()
```

Comment:
agree with @abhinav. I don't really see the reason to do anything about the exception here.
In general, it can be two reasons: 1) programmer error: the developer should fix their code. 2) host.json file gets corrupted: the program should stop right away, i.e., what's the point to continue if there is no way connect to Hyperbahn?